ui/src/app/components/copy-modal.tsx

```diff
@@ -232,6 +232,8 @@ class CopyModalComponent extends React.Component<Props, State> {
   }
 
   goToDestinationWorkspace() {
+    currentWorkspaceStore.next(null);
```

Comment:
This seems suspect - shouldn't this be handled somewhere more centrally during navigation? It is probably impossible guard every scenario where the user switches between active workspaces, especially since they can use browser history to introduce unexpected transitions.

Reply:
Hmm.. yeah there probably are a few other cases that are not covered. Although, I think this does cover a lot of the official UI flows since it's for all the copy to workspace workflows.

For some context on why this was added, it prevents the following bug.

1. User navigates somewhere
2. Components start rendering and fetching data but gets the workspace from the current store which hasn't been updated yet
3. Route event completes and current workspace store is updated based off of the url params info
4. Component rerender with new workspace fetched through the store

There's a brief window between steps 2 and 4 where the component shows data for the old workspace.

Reply:
After typing this out, I realized that the delay in steps 2-4 is due to the get workspace call and not due to delay in the url params value being emitted. I added a `currentWorkspaceStore.next(null)` call to the top of that function to null out the current workspace before the GET request is made and it looks like that works too.

https://github.com/all-of-us/workbench/blob/6e699ae16b73d32b20d9199aa791e56a344a7996/ui/src/app/pages/app/component.ts#L158